tests/by-util/test_pmap.rs

```diff
@@ -211,27 +241,116 @@ fn assert_extended_format(pid: u32, s: &str) {
+        assert!(re.is_match(line), "failing line: '{line}'");
     }
 
     let expected_separator = "---------------- ------- ------- ------- ";
-    assert_eq!(expected_separator, lines[line_count - 2]);
+    assert_eq!(
+        expected_separator,
+        lines[line_count - 2],
+        "failing line: '{}'",
+        lines[line_count - 2]
+    );
 
     let re = Regex::new(r"^total kB +[1-9][0-9]* +\d+ +\d+$").unwrap();
     assert!(
         re.is_match(lines[line_count - 1]),
-        "failing line: {}",
+        "failing line: '{}'",
+        lines[line_count - 1]
+    );
+}
+
+// Ensure `s` has the following more extended format (-X):
+//
+// 1234:   /some/path
+//          Address Perm           Offset    Device      Inode Size  Rss  Pss Pss_Dirty Referenced Anonymous LazyFree ShmemPmdMapped FilePmdMapped Shared_Hugetlb Private_Hugetlb Swap SwapPss Locked THPeligible Mapping
+// 000073eb5f4c7000 r-xp 0000000000036000 008:00008    2274176 1284 1148 1148         0       1148         0        0              0             0              0               0    0       0      0           0 ld-linux-x86-64.so.2
+// 00007ffd588fc000 r--p 0000000000000000 000:00000    2274176   20   20   20        20         20        20        0              0             0              0               0    0       0      0           0 [stack]
+// ffffffffff600000 rw-p 0000000000000000 000:00000    2274176   36   36   36        36         36        36        0              0             0              0               0    0       0      0           0 (one intentional trailing space)
+// ...
+//                                                             ==== ==== ==== ========= ========== ========= ======== ============== ============= ============== =============== ==== ======= ====== =========== (one intentional trailing space)
+//                                                             4164 3448 2826       552       3448       552        0              0             0              0               0    0       0      0           0 KB (one intentional trailing space)
+#[cfg(target_os = "linux")]
+fn assert_more_extended_format(pid: u32, s: &str) {
+    let lines: Vec<_> = s.lines().collect();
+    let line_count = lines.len();
+
+    let re = Regex::new(&format!("^{pid}:   .+[^ ]$")).unwrap();
+    assert!(re.is_match(lines[0]));
+
+    let re = Regex::new(r"^         Address Perm           Offset    Device      Inode +Size +Rss +Pss +Pss_Dirty +Referenced +Anonymous( +KSM)? +LazyFree +ShmemPmdMapped +FilePmdMapped +Shared_Hugetlb +Private_Hugetlb +Swap +SwapPss +Locked +THPeligible( +ProtectionKey)? +Mapping$").unwrap();
+    assert!(re.is_match(lines[1]), "failing line: '{}'", lines[1]);
+
+    let re = Regex::new(r"^[0-9a-f]{16} (-|r)(-|w)(-|x)(p|s) [0-9a-f]{16} [0-9a-f]{3}:[0-9a-f]{5} +\d+ +[1-9][0-9]* +\d+ +\d+ +\d+ +\d+ +\d+( +\d+)? +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+( +\d+)? (|\[[a-zA-Z ]+\]|[a-zA-Z0-9._-]+)$").unwrap();
```

Comment:
An underscore is also allowed in the `Mapping`. On my machine it fails on a `[vvar_vclock]` entry.
```suggestion
    let re = Regex::new(r"^[0-9a-f]{16} (-|r)(-|w)(-|x)(p|s) [0-9a-f]{16} [0-9a-f]{3}:[0-9a-f]{5} +\d+ +[1-9][0-9]* +\d+ +\d+ +\d+ +\d+ +\d+( +\d+)? +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+ +\d+( +\d+)? (|\[[a-zA-Z_ ]+\]|[a-zA-Z0-9._-]+)$").unwrap();
```